persistence/relational-jdbc/src/main/java/org/apache/polaris/persistence/relational/jdbc/models/ModelEntity.java

```diff
@@ -176,6 +202,14 @@ public static Builder builder() {
 
   @Override
   public PolarisBaseEntity fromResultSet(ResultSet r) throws SQLException {
+    // location_without_scheme column was added in schema version 2
+    String locationWithoutScheme = null;
+    try {
+      locationWithoutScheme = r.getString("location_without_scheme");
+    } catch (SQLException e) {
+      // Column does not exist, handle gracefully
+    }
```

Comment:
I think this part of logic would be simpler if we allow the object to have a new filed for schema version.
```
if (schemaVersion >= 2) {
     locationWithoutScheme = r.getString("location_without_scheme");
}
```